feat: ✨ add `create_raw_file_name()`

signekb opened the pull request:
## Description 

This PR adds the function `create_raw_file_name()` with a test. 
For this, I also added a helper function `get_compact_iso_timestamp()` that generates a compact version of the iso timestamp from `get_iso_timestamp(), by removing characters as ":", "T", and "-". It does, however, add a "-" between the data and time for readability.

Closes #802 

<!-- Select quick/in-depth as necessary -->
This PR needs an in-depth review.

## Checklist

- [X] Added or updated tests
- [X] Tests passed locally
- [X] Linted and formatted code

signekb committed:
feat: :sparkles: add helper function `get_compact_iso_timestap()` with tests

This function generates a compact version of the iso timestamp from `get_iso_timestamp(), by removing characters as ":", "T", and "-". It does add a "-" between the data and time for readability.

signekb committed:
feat: :sparkles: add `create_raw_file_name()` with a test

signekb committed:
test: :white_check_mark: add "expected" to test name

lwjohnst86 commented on sprout/core/get_iso_timestamp.py:
```diff
@@ -8,3 +9,12 @@ def get_iso_timestamp() -> str:
+    """Generates a compact timestamp.
+
+    Returns:
+        The timestamp as a string. E.g. `20240514-05000100`.
```
Good to convert timestamp to something that is more friendly, especially since some computers don't like `:` in the filename.

But, this isn't ISO 8601 compliant, e.g. lots of Stack Exchange posts on this topic of filenames and timestamps like [here](https://serverfault.com/questions/292014/preferred-format-of-file-names-which-include-a-timestamp/370766#370766). 

A format that adheres to ISO 8601 would be something like `yyyy-mm-ddThhmmssZ` or `yyyymmddThhmmssZ`. Both the `T` and `Z` are necessary to maintain compliance. The `Z` being to indicate UTC timezone, which we can for now set as the default (I don't think the exact time they uploaded will be an issue, but maybe?). I kinda prefer the one with the hyphens because it is a bit easier to read, but it mixes the ISO 8601 format a bit (without `:` but with `-`) so is less compliant the the one with no `-` or `:`. You decide which of the two you like.

signekb replied:
Oh, yes! You're right! Sorry about that. 
Good idea with just adding the `Z` for UTC. I also don't think that will be an issue; I would think that if anything happens, we'll be mostly concerned with "relative time" - e.g., which file was uploaded first/last. 
I'm 100 % for the `-` version 👍

lwjohnst86 reviewed (changes requested):
Nice :relaxed:

signekb committed:
fix: :adhesive_bandage: update `get_compact_iso_timestamp()` to return datetime following the iso format

signekb committed:
test: :white_check_mark: add check of whether the returned datetime is iso format

Using `datetime.fromisoformat()` that returns `ValueError` if the datetime string doesn't contain a valid ISO 8601 format.

signekb committed:
docs: :memo: update and minor edits to docstrings

signekb committed:
test: :white_check_mark: update to return filename containing datetime in valid iso format

lwjohnst86 reviewed (approved):
:grin:
